sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -368,7 +369,10 @@ abstract class UnixTime extends BinaryExpression with ExpectsInputTypes {
           t.asInstanceOf[Long] / 1000000L
         case StringType if right.foldable =>
           if (constFormat != null) {
-            Try(new SimpleDateFormat(constFormat.toString).parse(
+            if (formatter == null) {
+              formatter = Try(new SimpleDateFormat(constFormat.toString)).getOrElse(null)
```

Comment:
`.orNull` is available? but, this will just cause an NPE in the next line, which will also get squashed. Do we really want to hide both of these? why not let the error fly?